client/components/site-verticals-suggestion-search/index.jsx

```diff
@@ -105,14 +112,26 @@ export class SiteVerticalsSuggestionSearch extends Component {
 			// Don't trigger a search if there's already an exact, non-user-defined match from the API
 			! result
 		) {
-			this.props.requestVerticals( value, this.props.searchResultsLimit );
+			if ( valueLength < 5 ) {
+				this.requestVerticalsThrottled(
+					value,
+					this.props.searchResultsLimit,
+					this.setSearchResults
+				);
+			} else {
+				this.requestVerticalsDebounced(
+					value,
+					this.props.searchResultsLimit,
+					this.setSearchResults
+				);
+			}
 		}
 
 		this.setState( { searchValue: value } );
 		this.updateVerticalData( result, value );
 	};
 
-	getSuggestions = () => this.props.verticals.map( vertical => vertical.verticalName );
+	getSuggestions = () => this.state.results.map( vertical => vertical.verticalName );
 
 	sortSearchResults = ( suggestionsArray, queryString ) => {
 		let queryMatch;
```

Comment:
Thanks @niranjan-uma-shankar 

> Any reason why exact matches(in the variable queryMatch ) are pushed to the bottom of the dropdown? If the match has is_user_input_vertical true, then it's appropriate to push it down, but not otherwise

It's a good point. We just get an array of strings back from the child component, which renders the list, so without doing a lookup, we don't know it it's user input or not. 

The lookup is a little expensive, but doable. I think your observation provides more evidence that we should create a custom component for the vertical search API, that is, a component whose methods expect a specific data structure (a vertical object). I first [argued that in this PR.](https://github.com/Automattic/wp-calypso/pull/31405#issuecomment-472302718)

As it is, we moving between strings and arrays of objects, and have to perform constant array lookups to see if we're dealing with a valid API vertical.

> I think startsWithResults is not needed since we add an RTL compatible starts with word boundary check in the backend

Does the API take care of sorting now as well? If it does, then yes, we can probably get rid of it.

> I might be overseeing something here, but I'm not sure if we need sortSearchResults at all, unless it is solving a problem not addressed in the backend.

See previous comment. I suspect it's a hangover from the days when the topics were in the clientside bundle. [Introduced here](https://github.com/Automattic/wp-calypso/commit/9ac9ce667585095a0409a7e38814f967e8fd4a71#diff-c8525800f7c2200c5666fe5acb42f85eR113). 

Nice find. I'll add a note to cull it. 🔪 :)

Reply:
> I think startsWithResults is not needed since we add an RTL compatible starts with word boundary check in the backend

After some further testing, it looks like the sorting does help. Compare the following:

With the client sorting method:

<img width="223" alt="Screen Shot 2019-03-18 at 3 30 54 pm" src="https://user-images.githubusercontent.com/6458278/54506918-fb333200-4992-11e9-8af9-f3b64aa398be.png">

Without:

<img width="246" alt="Screen Shot 2019-03-18 at 3 30 05 pm" src="https://user-images.githubusercontent.com/6458278/54506915-f8384180-4992-11e9-8f7b-1cd99c0ca645.png">

At least for LTR languages, it's more logical to place results that start with what you're typing at the top.

Perhaps we could disable this for RTL languages until we test it properly.